src/java.base/share/classes/java/lang/StableValue.java

```diff
@@ -401,17 +402,13 @@
  * thread safe and guarantee at-most-once-per-input invocation.
  *
  * <h2 id="performance">Performance</h2>
- * A stable value that is <em>set</em> is treated as a constant by the JVM, enabling the
- * same performance optimizations that are available for {@code final} fields.
- * As such, stable values can be used to replace {@code final} fields in cases where
- * <em>at-most-once</em> update semantics is crucial, but where the eager initialization
- * semantics associated with {@code final} fields is too restrictive.
+ * The _content_ of a set stable value is treated as a constant by the JVM, provided that
+ * the reference to the stable value is also constant (e.g. in cases where the
+ * stable value itself is stored in a `static final` field).
  * <p>
- * In JDK 24, {@code final} instance fields in records and hidden classes (such as classes
- * spun from method references) are trusted by the JVM, allowing them to be treated as
- * constants. However, {@code final} instance fields in regular classes are <em>not</em>
- * trusted meaning that such fields are <em>not</em> eligible for performance
- * optimizations by the JVM (such as constant folding).
+ * This means that, at least in some cases, access to the content of a stable value
+ * enjoys the same constant-folding optimizations that are available when accessing
+ * `static final` fields.
```

Comment:
We probably need to extra para that says something about stable suppliers/functions etc. E.g. since these are cleanly built on top of SV, if these are constants (again, stored in `static final` field) their access can be constant-folded. E.g. make it clear that this property is compositional. It doesn't just hold for a single constant SV, but also for another constant objects that wraps a SV.